apps/api/app/resources.ts

```diff
@@ -33,20 +33,35 @@ router.get("/game/:game_name/:game_version/iframe", async (ctx) => {
     gameInfo?.viewer?.alternate?.url && ctx.query.alternate === "1" ? gameInfo?.viewer.alternate : gameInfo.viewer;
   const viewerUrl = ctx.query.customViewerUrl || viewer.url;
 
-  ctx.body = `
-    <html>
-      <head>
-        <meta charset="UTF-8">
-        ${viewer.dependencies.scripts.map((dep) => `<${"script"} src='${dep}'></${"script"}>`).join("\n")}
-        <${"script"} src='${viewerUrl}' type='text/javascript'> </${"script"}>
-        ${viewer.dependencies.stylesheets
-          .map((dep) => `<link type='text/css' rel='stylesheet' href='${dep}'></link>`)
-          .join("\n")}
+  const stylesheets = viewer.dependencies.stylesheets
+    .map((dep) => `<link type='text/css' rel='stylesheet' href='${dep}'></link>`)
+    .join("\n");
+  const scripts = viewer.dependencies.scripts.map((dep) => `<${"script"} src='${dep}'></${"script"}>`).join("\n");
+  const viewerScript = `<${"script"} src='${viewerUrl}' type='text/javascript'></${"script"}>`;
+
+  const template = viewer.topLevelVariable === "clash" ? `
+    <head>
+      <meta charset="UTF-8">
+      ${stylesheets}
       </head>
-      <body>
-        <div id='app'>
-        </div>
-      </body>
+    <body>
+      <canvas id='glcanvas' style="margin: 0; padding: 0; width: 100%; height: 100%; overflow: hidden; position: absolute; z-index: 0;"></canvas>
```

Comment:
This is specific to clash, for other games it would add a full-screen canvas

In the viewer for clash, you can add this directly with `body.append(...)` in the JS, in a synchronous manner

Reply:
This is only in the clash special logic